core/src/apps/ethereum/helpers.py

```diff
@@ -195,6 +195,22 @@ def format_ethereum_amount(
     return f"{amount} {suffix}"
 
 
+def get_account_and_path(address_n: list[int]) -> tuple[str | None, str | None]:
```

Comment:
This function has logical error. Firstly, we extract `address_n[1]` without checking for correct length. And then we extract `path` and `account_path` based on whether `address_n` is truthy or not.

The path (`address_n`) is probably checked in `paths.validate_path(...)` earlier. However, we can probably improve this function and apply some defensive programming. WDYT

```python
if not address_n or len(address_n) < 2:
     return (None, None)
```

Reply:
Indeed! https://github.com/trezor/trezor-firmware/pull/4175/commits/5b267d4e892ade8cc3908d512a62abb4b25c12b4#diff-c6fb97bf1b15b1422470f5df7175252de28b102f6e3022eb3cb797dfabe49e17R203